test/test_regression.py

```diff
@@ -0,0 +1,312 @@
+""" Regression tests for https://github.com/nameko/nameko/issues/428
```

Comment:
By "as a reference" I mean it would be nice to see all the different failures that this fix addresses, alongside the fix.

They're slow because several of them only fail after heartbeats are missed, and it can't realistically beat faster than once every few seconds.